Fixes #4294 : Added Profile delete message with AlertDialog

theayushyadav11 opened the pull request:
Fixes #4294 : This pull request introduces a new feature to notify users when a profile is successfully deleted. It includes the creation of a new dialog fragment, updates to existing fragments to integrate this new feature, and necessary string resources.

### New Feature Implementation:

* [`app/src/main/java/org/oppia/android/app/settings/profile/ProfileDeleteSuccessDialogFragment.kt`](diffhunk://#diff-2203d0609a904e5e457efadd299665f9c28d3f8bb4f8ef9d3479d872cd80a246R1-R62): Created a new `DeleteProfileSuccessDialogFragment` class to notify users after a profile is successfully deleted.

### Integration with Existing Components:

* [`app/src/main/java/org/oppia/android/app/fragment/FragmentComponentImpl.kt`](diffhunk://#diff-9de2bd314b90b27a64b5f0198a1e117d1401aca4c3bf6d41efdf0c0927d39b3cR66): Added import and injection for `DeleteProfileSuccessDialogFragment`. [[1]](diffhunk://#diff-9de2bd314b90b27a64b5f0198a1e117d1401aca4c3bf6d41efdf0c0927d39b3cR66) [[2]](diffhunk://#diff-9de2bd314b90b27a64b5f0198a1e117d1401aca4c3bf6d41efdf0c0927d39b3cR132)
* [`app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt`](diffhunk://#diff-12a41502c6dc988ca4dee178dfe871257b0be385b3453ed5aaef6bf16611c90fL154-R157): Updated to show the new `DeleteProfileSuccessDialogFragment` upon successful profile deletion.

### String Resources:

* [`app/src/main/res/values/strings.xml`](diffhunk://#diff-5e01f7d37a66e4ca03deefc205d8e7008661cdd0284a05aaba1858e6b7bf9103R348): Added a new string resource for the profile deletion success message.

### Test File Exemptions:

* [`scripts/assets/test_file_exemptions.textproto`](diffhunk://#diff-6ed782dd71126d847e7bac39eca30f830be55b72aa10b7e944612f2463003e24R2093-R2096): Added an exemption for the new `ProfileDeleteSuccessDialogFragment`.

### Video Demo
https://github.com/user-attachments/assets/d3a71f37-b6e5-43cb-b411-76af5f56042b 

### Video Demo on Tablet

https://github.com/user-attachments/assets/04c0912e-3dd2-45c8-bc3b-0025887394da

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

theayushyadav11 committed:
Added Profile delete message with AlertDialog

theayushyadav11 committed:
Fixed naming

theayushyadav11 committed:
removed and added newline

theayushyadav11 committed:
removed ktlint

theayushyadav11 committed:
removed spacing and unused import

theayushyadav11 committed:
changed name

theayushyadav11 committed:
changed name

theayushyadav11 commented:
@adhiamboperes PTAL.

theayushyadav11 commented:
Hey @adhiamboperes,
Could you please review my updates and let me know if any adjustments are needed? Thank you for your guidance!

adhiamboperes commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileDeleteSuccessDialogFragment.kt:
```diff
@@ -0,0 +1,61 @@
+
+    /** This function returns a new instance of [ProfileDeleteSuccessDialogFragment]. */
+    fun createNewInstance(): ProfileDeleteSuccessDialogFragment {
+      return ProfileDeleteSuccessDialogFragment()
```
```suggestion
    /** Returns a new instance of [ProfileDeleteSuccessDialogFragment]. */
    fun createNewInstance(): ProfileDeleteSuccessDialogFragment = ProfileDeleteSuccessDialogFragment()
```

adhiamboperes commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileDeleteSuccessDialogFragment.kt:
```diff
@@ -0,0 +1,61 @@
+  }
+
+  @Inject
+  lateinit var resourceHandler: AppLanguageResourceHandler
```
Move injected field to the top of the file.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileDeleteSuccessDialogFragment.kt:
```diff
@@ -0,0 +1,61 @@
+class ProfileDeleteSuccessDialogFragment : InjectableDialogFragment() {
+
+  companion object {
+    /** Argument key for Profile Deletion Success Dialog in [ProfileEditFragmentPresenter]. */
```
```suggestion
    /** Tag for [ProfileDeleteSuccessDialogFragment]. */
```

adhiamboperes commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileDeleteSuccessDialogFragment.kt:
```diff
@@ -0,0 +1,61 @@
+          resourceHandler.getStringInLocale(R.string.profile_edit_delete_successful_message)
+        )
+        setPositiveButton(
+          resourceHandler.getStringInLocale(R.string.log_out_dialog_okay_button)
```
It's better to create a screen specific string since we have future work splitting strings per screen. e.g. `profile_edit_delete_dialog_positive_button`

adhiamboperes commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt:
```diff
@@ -151,16 +149,11 @@ class ProfileEditFragmentPresenter @Inject constructor(
-              intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
-              fragment.startActivity(intent)
-            }
+            ProfileDeleteSuccessDialogFragment.createNewInstance()
+              .showNow(
+                fragment.childFragmentManager,
+                ProfileDeleteSuccessDialogFragment.DELETE_PROFILE_SUCCESS_DIALOG_FRAGMENT_TAG
+              )
```
Here is an example of a test that verifies a dialog is shown -> https://github.com/oppia/oppia-android/blob/2f68639f90e84fde029b92da48a9bc925cea7cc8/app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt#L706. Please add a test for this change in `ProfileEditFragmentTest`.

adhiamboperes reviewed (changes requested):
Thanks @theayushyadav11! 

I have left some suggestions. PTAL. 

Also please add a tablet screen recording(without talkback), since tablet behaviour is also affected.

theayushyadav11 committed:
Added test

theayushyadav11 commented:
Thank you so much for your detailed review, @adhiamboperes. I’ve worked on addressing all the comments in my recent commit. If there are any areas that still need improvement, I’d be grateful if you could guide me further. Looking forward to your feedback!

### Video Demo on Tablet

https://github.com/user-attachments/assets/04c0912e-3dd2-45c8-bc3b-0025887394da

theayushyadav11 committed:
fixed tests

theayushyadav11 commented:
Hey @adhiamboperes PTAL.

TanishMoral11 commented:
I would like to suggest changing the text color of 'Delete' to red ( same as Profile Deletion )  so that it looks more visually appealing.

adhiamboperes commented:
Thanks @theayushyadav11! 

Could you please check that the tests pass?

```
org.oppia.android.app.settings.profile.ProfileEditFragmentTest > testProfileEdit_startWithUserProfile_clickDelete_configChange_checkOpensDeletionDialog_checkOpensSuccessDialogIsVisible FAILED
    androidx.test.espresso.NoMatchingRootException: Matcher 'is dialog' did not match any of the following roots: [Root{application-window-token=android.view.ViewRootImpl$W@771eb3f3, window-token=android.view.ViewRootImpl$W@771eb3f3, has-window-focus=true, layout-params-type=1, layout-params-string={(0,0)(fillxfill) ty=BASE_APPLICATION wanim=0x7f120005
      fl=LAYOUT_IN_SCREEN LAYOUT_INSET_DECOR SPLIT_TOUCH HARDWARE_ACCELERATED DRAWS_SYSTEM_BAR_BACKGROUNDS
      pfl=FORCE_DRAW_STATUS_BAR_BACKGROUND FIT_INSETS_CONTROLLED
      fitSides=}, decor-view-string=DecorView{id=-1, visibility=VISIBLE, width=470, height=320, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params={(0,0)(fillxfill) ty=BASE_APPLICATION wanim=0x7f120005
      fl=LAYOUT_IN_SCREEN LAYOUT_INSET_DECOR SPLIT_TOUCH HARDWARE_ACCELERATED DRAWS_SYSTEM_BAR_BACKGROUNDS
      pfl=FORCE_DRAW_STATUS_BAR_BACKGROUND FIT_INSETS_CONTROLLED
      fitSides=}, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=1}}]
 
```

theayushyadav11 committed:
Merge branch 'develop' into delete-profile-message

theayushyadav11 commented:
Thanks for the review @adhiamboperes ,Updating it in my next commit.

theayushyadav11 committed:
Fixed failed tests

theayushyadav11 committed:
Merge remote-tracking branch 'origin/delete-profile-message' into delete-profile-message

theayushyadav11 committed:
cleanup

theayushyadav11 commented:
Hey @adhiamboperes PTAL.

theayushyadav11 committed:
Merge branch 'develop' into delete-profile-message

adhiamboperes commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileDeleteSuccessDialogFragment.kt:
```diff
@@ -0,0 +1,59 @@
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import javax.inject.Inject
+
+/** [DialogFragment] that notifies the user after a profile is successfully deleted. */
```
You need to do ctrl + alt + L to r5eformat this class and resolve the max line exceeded ktlint error.

adhiamboperes commented on app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt:
```diff
@@ -167,34 +167,46 @@ class ProfileEditFragmentTest {
 
   @Test
-  fun testProfileEdit_startWithUserProfile_clickProfileDeletionButton_checkOpensDeletionDialog() {
+  fun testProfileEdit_startWithUserProfile_clickProfileDeletionButton_checkOpensDeletionDialog_checkOpensSuccessDialog() {
```
```suggestion
  fun testProfileEdit_clickProfileDeletionButton_checkOpensDeletionDialog_checkOpensSuccessDialog() {
```

adhiamboperes commented on app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt:
```diff
@@ -167,34 +167,46 @@ class ProfileEditFragmentTest {
   @Test
   @Config(qualifiers = "land")
-  fun testProfileEdit_configChange_startWithUserProfile_clickDelete_checkOpensDeletionDialog() {
+  fun testProfileEdit_configChange_startWithUserProfile_clickDelete_checkOpensDeletionDialog_checkOpensSuccessDialog() {
```
```suggestion
  fun testProfileEdit_configChange_clickDelete_checkOpensDeletionDialog_checkOpensSuccessDialog() {
```

adhiamboperes commented on app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt:
```diff
@@ -167,34 +167,46 @@ class ProfileEditFragmentTest {
   @Test
   @Config(qualifiers = "land")
-  fun testProfileEdit_startWithUserProfile_clickDelete_configChange_checkDeletionDialogIsVisible() {
+  fun testProfileEdit_startWithUserProfile_clickDelete_configChange_checkOpensDeletionDialog_checkOpensSuccessDialogIsVisible() {
```
```suggestion
  fun testProfileEdit_clickDelete_landscapeMode_checkOpensDeletionDialog() {
```

adhiamboperes reviewed (changes requested):
Thanks @theayushyadav11! Nice work on this PR.

There are some ktlin failures preventing merge, and I have added some suggestions on how to fix them. PTAL.

theayushyadav11 committed:
Reformated files

theayushyadav11 committed:
Merge remote-tracking branch 'origin/delete-profile-message' into delete-profile-message

theayushyadav11 committed:
Reformated files

theayushyadav11 committed:
Reformated files

theayushyadav11 committed:
Reformated files

theayushyadav11 commented:
Thanks @adhiamboperes ,PTAL 
Can You please suggest me a better way to check Ktlint errors because the command ` java -jar /home/runner/oppia-android-tools/ktlint -F --android domain/src/**/*.kt utility/src/**/*.kt data/src/**/*.kt app/src/**/*.kt testing/src/**/*.kt scripts/src/**/*.kt  instrumentation/src/**/*.kt.` shows "cannot access jar file error" and I have to manually check for it.

adhiamboperes reviewed (approved):
Thanks @theayushyadav11! This LGTM.

adhiamboperes commented:
@theayushyadav11, you may be experiencing the jar error because your ktlint was not properly set up.  Please re-run `bash scripts/setup.sh`. You should see the message, "Ktlint file downloaded."

adhiamboperes committed:
Merge branch 'develop' into delete-profile-message
